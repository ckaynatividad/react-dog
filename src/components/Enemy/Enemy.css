.detail {
  color: rgba(255, 255, 255, 0.884);
}

button {
  margin-top: 100px;
  width: 80px;
  height: 20px;
  font-size: 6.4pt;
  font-style: italic;
  background-color: rgba(0, 128, 128, 0.815);
}
