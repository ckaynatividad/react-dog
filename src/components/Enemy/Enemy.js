import React from 'react';
import { Link } from 'react-router-dom';
import './Enemy.css';

export default function EnemyDetail(props) {
  
  return (
    <div className="detail">
      <h1>{props.name}</h1>
      <img src={props.image} />
      <p>level: {props.level}</p>
      <p>type: {props.type}</p>
      <p>element: {props.element} </p>

      <Link to={`/${props.id}/edit`}>Edit</Link>
      <button onClick={props.handleDelete}>delete enemy</button>

    </div>
  );
}
