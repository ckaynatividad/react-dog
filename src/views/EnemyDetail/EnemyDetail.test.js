import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router';
import Enemy from './EnemyDetail';

it('should render enemy info', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/8']}>
      <Route path="/:id" component={Enemy} />
    </MemoryRouter>
  );
  await screen.findByText('ginseng jar');
  expect(container).toMatchSnapshot();
});