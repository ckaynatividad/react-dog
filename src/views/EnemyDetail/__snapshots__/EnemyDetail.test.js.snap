// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render enemy info 1`] = `
<div>
  <div>
    <div
      class="detail"
    >
      <h1>
        ginseng jar
      </h1>
      <img
        src="https://static.wikia.nocookie.net/maplestory/images/a/a9/Mob_Ginseng_Jar.png"
      />
      <p>
        level: 
        139
      </p>
      <p>
        type: 
        enchanted
      </p>
      <p>
        element: 
        neutral
         
      </p>
      <a
        href="/8/edit"
      >
        Edit
      </a>
      <button>
        delete enemy
      </button>
    </div>
  </div>
</div>
`;
